lib/iris/tests/idiff.py

```diff
@@ -25,17 +25,52 @@
 from __future__ import (absolute_import, division, print_function)
 from six.moves import (filter, input, map, range, zip)  # noqa
 
+import argparse
+import codecs
+import contextlib
+import hashlib
+import json
 import os.path
+import requests
 import shutil
 import sys
+import warnings
 
+import filelock
 import matplotlib.pyplot as plt
 import matplotlib.image as mimg
+import matplotlib.testing.compare as mcompare
 import matplotlib.widgets as mwidget
 
+# Force iris.tests to use the ```tkagg``` backend by using the '-d'
+# command-line argument as idiff is an interactive tool that requires a
+# gui interface.
+sys.argv.append('-d')
+import iris.tests
+import iris.util as iutil
 
-def diff_viewer(expected_fname, result_fname, diff_fname):
-    plt.figure(figsize=(16, 16))
+
+_HASH_DIR = os.path.join('results', 'visual_tests')
```

Comment:
this folder is under source control, In don't think any code should be writing images to a folder under iris source control.  i think the potential for new image files to get added by mistake is too high

could hashed named image files be put elsewhere please?
they'll need to be moved by a developer again anyway